### DISCUSSION TITLE
Error when saving a double in DataGrid

### DISCUSSION THREAD

**Discussion by zawarq** [Megabit/Blazorise#5214] [General]
I have a DataGrid with the following column definition (amongst others):

<DataGridColumn Field="@nameof(PatientLabTest.TestValue)" Caption="Value" Editable Filterable="false" />

PatientLabTest.TestValue has type double. The data loads from server. When I create a new record, or edit an existing record whereby I change the TestValue, I get the following error in the browser console:

> Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
>       Unhandled exception rendering component: Specified cast is not valid.
> System.InvalidCastException: Specified cast is not valid.
>    at System.Object.lambda_method96(Closure , PatientLabTest , Object )
>    at Blazorise.DataGrid.DataGridColumn`1[[NutritionProViewModels.Practice.PatientLabTest, NutritionProViewModels, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetValue(PatientLabTest item, Object value)
>    at Blazorise.DataGrid.DataGrid`1[[NutritionProViewModels.Practice.PatientLabTest, NutritionProViewModels, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].SetItemEditedValues(PatientLabTest item)
>    at Blazorise.DataGrid.DataGrid`1.<SaveItem>d__76[[NutritionProViewModels.Practice.PatientLabTest, NutritionProViewModels, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext()
>    at Blazorise.DataGrid.DataGrid`1.<SaveInternal>d__74[[NutritionProViewModels.Practice.PatientLabTest, NutritionProViewModels, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext()
>    at Blazorise.DataGrid._BaseDataGridRowEdit`1.<Save>d__12[[NutritionProViewModels.Practice.PatientLabTest, NutritionProViewModels, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null]].MoveNext()
>    at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
>    at Blazorise.Button.ClickHandler(MouseEventArgs eventArgs)
>    at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
>    at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)

The above error occurs before I hit my OnRowInsert/Update handlers.

If I edit an existing record, and saving it **without** changing the value of TestValue, it works fine.

Note this this datagrid was working fine when I had Blazorise version 1.2, but when I upgraded to the latest 1.4.0 this error started appearing.

Any help will be greatly appreciated :)

Regards
Zawar

**Comment by David-Moreira** [comment]
Hello @zawarq 
Can you post the code? Thanks!

**Reply by zawarq** [reply]
Hi @David-Moreira 

Yes, certainly. Attached please find.
[PatientLabTest.cs.txt](https://github.com/Megabit/Blazorise/files/13867178/PatientLabTest.cs.txt)
[PatientLabTests.razor.txt](https://github.com/Megabit/Blazorise/files/13867179/PatientLabTests.razor.txt)

**Reply by David-Moreira** [reply]
Hello, It does seem to be a bug we've mistakenly introduced in 1.4. The `DataGridColumn` is not figuring out the correct value type for numeric types. 
You should be able to use a `DataGridNumericColumn` instead while we work on the fix.
I'll go ahead and open an issue.

**Comment by zawarq** [comment]
Thanks David! I'll look into DataGridNumericColumn now

Regards
Zawar

**Reply by David-Moreira** [reply]
Sure. In the meantime I forgot to link the related issue. 
https://github.com/Megabit/Blazorise/issues/5217

It's been fixed. And should be available on the next release.
